src/libxrpl/protocol/STParsedJSON.cpp

```diff
@@ -373,10 +374,44 @@ parseLeaf(
             {
                 if (value.isString())
                 {
-                    ret = detail::make_stvar<STUInt32>(
-                        field,
-                        beast::lexicalCastThrow<std::uint32_t>(
-                            value.asString()));
+                    if (field == sfPermissionValue)
+                    {
+                        std::string const strValue = value.asString();
+                        auto const granularPermission =
+                            Permission::getInstance().getGranularValue(
+                                strValue);
+                        if (!granularPermission)
+                        {
+                            // if it's not granular permission, parse as
+                            // transaction type permission.
+                            if (Permission::getInstance().isProhibited(
+                                    strValue))
```

Comment:
This is the wrong place to check if a transaction is supported. Transactions that come in from peers are never `Json`, and thus are never parsed as such. Neither, I think, is a transaction submitted as a `tx_blob`. Just parse the `TxType` here and do the check in the `Transactor`s. (See my comment on `isProhibited`.)